TESTING.asciidoc

```diff
@@ -76,7 +76,177 @@ In order to start a node with a trial license execute the following command:
 This enables security and other paid features and adds a superuser with the username: `elastic-admin` and
 password: `elastic-password`.
 
-==== Other useful arguments
+
+== Running verification tasks
+
+To run all verification tasks, including static checks, unit tests, and integration tests:
+
+---------------------------------------------------------------------------
+./gradlew check
+---------------------------------------------------------------------------
+
+Note that this will also run the unit tests and precommit tasks first. If you want to just
+run the integration tests (because you are debugging them):
+
+---------------------------------------------------------------------------
+./gradlew integTest
+---------------------------------------------------------------------------
+
+If you want to just run the precommit checks:
+
+---------------------------------------------------------------------------
+./gradlew precommit
+---------------------------------------------------------------------------
+
+Some of these checks will require `docker-compose` installed for bringing up
+test fixtures. If it's not present those checks will be skipped automatically.
+
+== How to write good tests?
+
+=== Test pyramid
```

Comment:
I'm not sure how appropriate this is here. It's not really Elasticsearch specific and starts to cross into opinions on testing which I think there are better places to have such discussions. If we have discrete conventions on these things that fine but I'm not sure we do here.

Reply:
I agree I don't think this really belongs here. IMO the statement below about unit tests being preferred is sufficient.